File: ChaeJangWoo/week_2/boj_11328.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>
using namespace std;

int main()
{
    int n;
    cin >> n;
    vector<string> result(n);
    for (int i = 0; i < n; i++)
    {
        string s1, s2;
        cin >> s1 >> s2;
        sort(s1.begin(), s1.end());
        sort(s2.begin(), s2.end());
        if (s1 == s2) { 
            result[i] = "Possible"; // 결과 저장
        } else {
            result[i] = "Impossible"; // 결과 저장
        }
    }

    for(int i = 0; i < n; i++){
        cout << result[i] << endl;
    }
}

File: KimJiYu/week_2/boj_11328.java
package nestnet_algorithm_2024_2_beginner.KimJiYu.week_2;

import java.util.Arrays;
import java.util.Scanner;

/*
[11328] Strfry
 */
public class boj_11328 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        // 테스트 케이스의 수
        int n = in.nextInt();
        in.nextLine();

        while (n-- > 0) {
            String str1 = in.next();
            String str2 = in.next();

            if (str1.length() != str2.length()) {
                System.out.println("Impossible");
                continue;
            }

            // 문자 배열로 전환
            char[] chArr1 = str1.toCharArray();
            char[] chArr2 = str2.toCharArray();

            // 알파벳순으로 정렬
            Arrays.sort(chArr1);
            Arrays.sort(chArr2);

            if (Arrays.equals(chArr1, chArr2)) {
                System.out.println("Possible");
            } else {
                System.out.println("Impossible");
            }
        }
        in.close();
    }
}